subprojects/model-core/src/main/java/org/gradle/api/internal/provider/ValueSupplier.java

```diff
@@ -286,6 +333,68 @@ public String toString() {
         }
     }
 
+    class EmptySideEffect<T> implements SideEffect<T> {
+
+        public static boolean isEmpty(@Nullable SideEffect<?> sideEffect) {
+            return sideEffect == null || sideEffect == INSTANCE;
+        }
+
+        private static <T> SideEffect<T> value() {
```

Comment:
I'd normally call such a method `instance()` or `getInstance()`. Is `value()` used somewhere else?